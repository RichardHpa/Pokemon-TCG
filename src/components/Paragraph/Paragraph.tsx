import { forwardRef } from 'react'
import clsx from 'clsx'

export type ParagraphProps = {
  className?: string
} & React.ComponentPropsWithoutRef<'p'>

export const Paragraph = forwardRef<HTMLParagraphElement, ParagraphProps>(
  ({ className, ...props }, ref) => {
    return (
      <p
        ref={ref}
        className={clsx('mb-3 text-gray-500 dark:text-gray-400', className)}
        {...props}
      />
    )
  },
)

Paragraph.displayName = 'Paragraph'
