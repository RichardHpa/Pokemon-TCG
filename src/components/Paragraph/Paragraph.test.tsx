import { render, screen } from '@testing-library/react'

import { Paragraph } from './Paragraph'

describe('Paragraph', () => {
  test('renders basic paragraph', () => {
    const { container } = render(<Paragraph>content</Paragraph>)
    expect(screen.getByText('content')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
