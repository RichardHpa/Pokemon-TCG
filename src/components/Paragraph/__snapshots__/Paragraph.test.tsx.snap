// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders basic paragraph 1`] = `
<div>
  <p
    class="mb-3 text-gray-500 dark:text-gray-400"
  >
    content
  </p>
</div>
`;
